sklearn/preprocessing/data.py

```diff
@@ -1182,6 +1182,35 @@ def powers_(self):
         return np.vstack(np.bincount(c, minlength=self.n_input_features_)
                          for c in combinations)
 
+    def get_feature_names(self, input_features=None):
```

Comment:
I've been suggesting for a while that this is an appropriate way to deal with feature names in extractor/transformer pipelines. I'm happy to see this.